src/tiledbsoma_ml/pytorch.py

```diff
@@ -438,13 +438,16 @@ def _mini_batch_iter(
                         if self.return_sparse_X
                         else X_io_batch[iob_idx : iob_idx + mini_batch_size].toarray()
                     )
+                    obs_batch = obs_io_batch.iloc[iob_idx : iob_idx + mini_batch_size]
+                    # Give `obs_batch` a fresh [0, N) index (instead of an out-of-context slice of the IO batch's index)
+                    obs_batch.index = pd.RangeIndex(len(obs_batch))
```

Comment:
why not the (slightly) more readable:
> obs_batch = obs_io_batch.iloc[iob_idx : iob_idx + mini_batch_size].reset_index(drop=True)

I'm fine w/ either, but this seems much more explicit (to the point that the comment isn't really necessary)

Reply:
Agreed, done.